File: Eunleelee/2024-01/1월 31일/햄버거_만들기.py
def solution(ingredient):
    answer = 0
    stack = []

    for i in ingredient:
        stack.append(i)
        if len(stack) >= 4 and stack[-1] == 1 and stack[-2] == 3 and stack[-3] == 2 and stack[-4] == 1:
            stack.pop()
            stack.pop()
            stack.pop()
            stack.pop()
            answer += 1
       
    return answer

File: yht0827/1월/1월 31일/햄버거 만들기.java
public int solution(int[] ingredient) {
	int[] newIngredient = new int[ingredient.length];
	int idx = 0, answer = 0;
	for (int ing : ingredient) {
		newIngredient[idx] = ing;
		idx++;

		if (idx >= 4) {
			if (newIngredient[idx - 4] == 1 && newIngredient[idx - 3] == 2 && newIngredient[idx - 2] == 3
				&& newIngredient[idx - 1] == 1) {
				idx -= 4;
				answer++;
			}
		}
	}

	return answer;
}